src/tools/perf/lib/libperf.c

```diff
@@ -1053,11 +1098,14 @@ static ucs_status_t ucp_perf_test_receive_remote_data(ucx_perf_context_t *perf,
 
     /* Initialize all endpoints and rkeys to NULL to handle error flow */
     for (i = 0; i < thread_count; i++) {
-        perf->ucp.tctx[i].perf.ucp.ep   = NULL;
-        perf->ucp.tctx[i].perf.ucp.rkey = NULL;
+        perf->ucp.tctx[i].perf.ucp.ep             = NULL;
+        perf->ucp.tctx[i].perf.ucp.rkey           = NULL;
+        perf->ucp.tctx[i].perf.ucp.self_ep        = NULL;
+        perf->ucp.tctx[i].perf.ucp.self_send_rkey = NULL;
+        perf->ucp.tctx[i].perf.ucp.self_recv_rkey = NULL;
```

Comment:
Seems that this should be in `ucp_perf_test_setup_self_endpoints`.

Reply:
`ucp_perf_test_setup_self_endpoints` is being called conditionally, we would want those attributes to be NULL otherwise, so in destruction, we will be able to check whether we should release those eps and rkeys.